File: android/app/src/main/java/org/fossasia/openevent/api/Urls.java
package org.fossasia.openevent.api;

import android.webkit.URLUtil;

import org.fossasia.openevent.utils.Utils;

/**
 * User: championswimmer
 * Date: 23/5/15
 */
public abstract class Urls {

    public static final String API_VERSION = "v1";

    /**
     * Change EVENT Id Here *
     */
    public static final int EVENT_ID = 1;

    public static final String WEB_APP_URL_BASIC = "http://fossasia.github.io/open-event-webapp/#/";

    public static final String EVENT = "event";

    public static final String SPEAKERS = "speakers";

    public static final String TRACKS = "tracks";

    public static final String SESSIONS = "sessions";

    public static final String SPONSORS = "sponsors";

    public static final String BOOKMARKS = "bookmarks";

    public static final String MICROLOCATIONS = "microlocations";

    public static final String SESSION_TYPES = "session_types";

    public static final String MAP = "map";

    public static String BASE_URL = "https://eventyay.com/api/v1/events/6";

    public static String FACEBOOK_BASE_URL = "https://graph.facebook.com";

    public static final String BASE_GET_URL = BASE_URL + "/api/" + API_VERSION;

    public static final String BASE_GET_URL_ALT = "https://raw.githubusercontent.com/fossasia/open-event/master/testapi/";

    // Replace the template in getAppLink() if changing it
    public static final String APP_LINK = "https://app_link_goes_here.com";

    public static final String INVALID_LINK = "http://abc//";

    public static final String EMPTY_LINK = "http://xyz//";

    public static final String GOOGLE_PLAY_HOME = "https://play.google.com/store";


    public static String getBaseUrl() {
        return BASE_URL;
    }

    public static void setBaseUrl(String baseUrl) {
        if (URLUtil.isValidUrl(baseUrl)) {
            if (!baseUrl.endsWith("/")) {
                BASE_URL = baseUrl + "/";
            } else {
                BASE_URL = baseUrl;
            }
        } else {
            if (!Utils.isEmpty(baseUrl)) {
                BASE_URL = INVALID_LINK;
            } else {
                BASE_URL = EMPTY_LINK;
            }
        }
    }

    /**
     * Checks if the app link is replaced by the generator, if not
     * returns null which is to be checked by caller to make decision
     * accordingly.
     * @return String
     */
    public static String getAppLink() {
        return APP_LINK.equals("https://app_link_goes_here.com")?GOOGLE_PLAY_HOME:APP_LINK;
    }
}

File: android/app/src/main/java/org/fossasia/openevent/common/api/Urls.kt
package org.fossasia.openevent.common.api

import android.webkit.URLUtil

import org.fossasia.openevent.common.utils.Utils

object Urls {

    const val API_VERSION = "v1"

    /**
     * Change EVENT Id Here *
     */
    const val EVENT_ID = 1

    const val WEB_APP_URL_BASIC = "http://fossasia.github.io/open-event-webapp/#/"

    const val EVENT = "event"

    const val SPEAKERS = "speakers"

    const val TRACKS = "tracks"

    const val SESSIONS = "sessions"

    const val SPONSORS = "sponsors"

    const val BOOKMARKS = "bookmarks"

    const val MICROLOCATIONS = "microlocations"

    const val SESSION_TYPES = "session_types"

    const val MAP = "map"

    @JvmField
    var BASE_URL = "https://eventyay.com/api/v1/events/6"

    const val FACEBOOK_BASE_URL = "https://graph.facebook.com"

    const val LOKLAK_BASE_URL = "https://api.loklak.org"

    @JvmField
    val BASE_GET_URL = "$BASE_URL/api/$API_VERSION"

    const val BASE_GET_URL_ALT = "https://raw.githubusercontent.com/fossasia/open-event/master/testapi/"

    // Replace the template in getAppLink() if changing it
    private const val APP_LINK = "https://app_link_goes_here.com"

    const val INVALID_LINK = "http://abc//"

    const val EMPTY_LINK = "http://xyz//"

    private const val GOOGLE_PLAY_HOME = "https://play.google.com/store"

    var baseUrl: String
        @JvmStatic
        get() = BASE_URL
        @JvmStatic
        set(baseUrl) = if (URLUtil.isValidUrl(baseUrl)) {
            BASE_URL = if (!baseUrl.endsWith("/")) {
                "$baseUrl/"
            } else {
                baseUrl
            }
        } else {
            BASE_URL = if (!Utils.isEmpty(baseUrl)) {
                INVALID_LINK
            } else {
                EMPTY_LINK
            }
        }

    /**
     * Checks if the app link is replaced by the generator, if not
     * returns null which is to be checked by caller to make decision
     * accordingly.
     * @return String
     */
    val appLink: String
        @JvmStatic
        get() = if (APP_LINK == "https://app_link_goes_here.com") GOOGLE_PLAY_HOME else APP_LINK

}
